### ISSUE TITLE
zIndex not working when using with reducers

### ISSUE THREAD

**Issue by suhdonghwi** [jacomyal/sigma.js#1131]
<img width="689" alt="image" src="https://user-images.githubusercontent.com/8275026/126055489-2c65efbf-48cd-48ba-ba1f-8e63b5c0d7f0.png">

I enabled zIndex feature by setting `zIndex: true` in renderer, and added `z` attribute to each nodes & edges. But like in the picture above, zIndex is not properly rendered.

**Comment by AVermeij** [comment]
Working as expected here - do you change the ``z`` attribute for the edges in an ``enterNode`` or ``clickNode`` event? This works for me:

```
renderer.on('clickNode', e=>{
  // Get focal node's edges and add to a map
  let node = e.node;
  let edgesMap = new Map();
  renderer.graph.edges(node).forEach(edge=>{edgesMap.set(edge, true)});

  renderer.graph.edges().forEach(edge=>{
    if (edgesMap.has(edge)) {
      renderer.graph.setEdgeAttribute(edge, 'color', '#ff0000');
      renderer.graph.setEdgeAttribute(edge, 'z', 1);
    } else {
      renderer.graph.setEdgeAttribute(edge, 'color', '#eee');
      renderer.graph.setEdgeAttribute(edge, 'z', 0);
    }
  });
});
```

In an accompanying ``clickStage`` event (or ``leaveNode`` when you use an ``enterNode`` event), you can then revert all edges back to their original colors and z-index.

**Comment by suhdonghwi** [comment]
Yes in that case z-index works, but it seems like that the real problem occurs when using `z` attribute with **reducers**. Applying `z` attribute in node or edge reducer does not have any effect to renderer.

Personally I [forked sigma.js and fixed](https://github.com/suhdonghwi/sigma.js/commit/a00b832aa68ce36b0ca5ae940a10b0b49e069f57) for my own project, but later I found [this pull request](https://github.com/jacomyal/sigma.js/pull/1080) which fixes the same problem. I hope it will be merged soon.

**Comment by AVermeij** [comment]
Not using reducers here, so can't really comment on that. Perhaps change the title and description of your issue to more accurately reflect that the issue occurs when using reducers?

**Comment by Yomguithereal** [comment]
Hello everyone, I see the issue with reducers and the linked PR indeed has the good insight to fix the issue. We will integrate this in the next beta release of the v2 I think.

**Comment by jacomyal** [comment]
Hi all,

I just pushed a proper fix to v2, mostly based on @abhikr713 [PR #1080](https://github.com/jacomyal/sigma.js/pull/1080), it will be available very soon in `v2.0.0-beta9`.
